modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogServiceImpl.java

```diff
@@ -877,4 +886,24 @@ private static void validateAlterTableColumn(
             throwUnsupportedDdl("Cannot decrease precision to {} for column '{}'.", target.precision(), origin.name());
         }
     }
+
+    private static CreateHashIndexParams createPkIndexParams(CreateTableParams params) {
+        return CreateHashIndexParams.builder()
```

Comment:
Maybe I'm missing something, but where's the index state?
PK index must be created as an RW index, but such enum/classification doesn't seem to exist. Are we planning to implement it later?

Reply:
About the state of indexes will be continued in epic https://issues.apache.org/jira/browse/IGNITE-17766.